internal/security/secretstore/vmkencryption_test.go

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package secretstore
+
+import (
+	"errors"
+	"testing"
+
+	. "github.com/edgexfoundry/edgex-go/internal/security/kdf/mocks"
+	. "github.com/edgexfoundry/edgex-go/internal/security/pipedhexreader/mocks"
+	"github.com/edgexfoundry/edgex-go/internal/security/secretstoreclient"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	. "github.com/edgexfoundry/go-mod-secrets/pkg/token/fileioperformer/mocks"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestVMKEncryptionNoIkm tests the no-op path
+func TestVMKEncryptionNoIkm(t *testing.T) {
+	// Arrange
+	fileOpener := &MockFileIoPerformer{}
+	pipedHexReader := &MockPipedHexReader{}
+	lc := &logger.MockLogger{}
+	kdf := &MockKeyDeriver{}
+
+	// Act / Assert
+	vmkEncryption := NewVMKEncryption(fileOpener, pipedHexReader, lc, kdf)
+	encrypting := vmkEncryption.IsEncrypting()
+	assert.False(t, encrypting)
+
+	vmkEncryption.WipeIKM()
+
+	fileOpener.AssertExpectations(t)
+	pipedHexReader.AssertExpectations(t)
+	kdf.AssertExpectations(t)
+}
+
+// TestVMKEncryption tests the happy path
+func TestVMKEncryption(t *testing.T) {
+	// Arrange
+	fakeIkm := make([]byte, 512)
+	fileOpener := &MockFileIoPerformer{}
+	pipedHexReader := &MockPipedHexReader{}
+	pipedHexReader.On("ReadHexBytesFromExe", "/bin/myikm", []string{}).Return(fakeIkm, nil)
+	lc := &logger.MockLogger{}
+	kdf := &MockKeyDeriver{}
+	kdf.On("DeriveKey", make([]byte, 512), uint(32), "vault0").Return(make([]byte, 32), nil)
+	kdf.On("DeriveKey", make([]byte, 512), uint(32), "vault1").Return(make([]byte, 32), nil)
+	initialInitResp := secretstoreclient.InitResponse{
+		Keys:       []string{"aabbcc", "ddeeff"},
+		KeysBase64: []string{"qrvM", "3e7/"},
+	}
+	initResp := initialInitResp
+
+	// Act & Assert
+	vmkEncryption := NewVMKEncryption(fileOpener, pipedHexReader, lc, kdf)
+	err := vmkEncryption.LoadIKM("/bin/myikm")
+	assert.NoError(t, err)
```

Comment:
Use `require.NoError()` to stop test here if fails. Do this in other locations and other tests where appropriate.